Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM alpine
+
+# Setup apt
+RUN apk -U --no-cache add \
+               build-base \
+               git \
+               linux-headers \
+               python3 \
+               python3-dev && \ 
+
+# Setup Snare
+    git clone --depth=1 https://github.com/mushorg/snare /opt/snare && \
+    cd /opt/snare/ && \
+    pip3 install --no-cache-dir --upgrade pip setuptools && \
+    pip3 install --no-cache-dir -r requirements.txt && \
+    python3.6 clone.py --target http://example.com && \
```

Comment:
It's better to use ARG here.
https://docs.docker.com/engine/reference/builder/#arg

Reply:
In your opinion, what is the advantage of using ARGs here? You'd have to rebuild anyway using a `build-arg list`, hence writing it directly to the `Dockerfile` seemed the obvious choice.
In a matter of using variables one could argue to remove `line 16` and incorporate it in the `CMD` while setting ENVs in the `docker-compose.yml`:

```
CMD python3.6 clone.py --target $TARGET && exec /usr/bin/python3.6 /opt/snare/snare.py --no-dorks --auto-update false --host-ip 0.0.0.0 --port 80 --page-dir $TARGET_NAME
```

Reply:
Good idea about ENV!
Then it would be better to check the $TARGET folder before.
I'd suggest something like this for the entrypoint script
```sh
#!/bin/sh

PYTHON=python3.6

if [ ! -d "$TARGET" ]; then
  $PYTHON clone.py --target $TARGET
fi

$PYTHON snare.py --no-dorks --auto-update false --host-ip 0.0.0.0 --port 80 --page-dir $TARGET

```